cmd/diagcfg/telemetry.go

```diff
@@ -36,11 +38,19 @@ func init() {
 	telemetryCmd.AddCommand(telemetryNameCmd)
 	telemetryCmd.AddCommand(telemetryEndpointCmd)
 
+	telemetryCmd.PersistentFlags().StringVarP(&dataDir, "datadir", "d", "", "Data directory for the node")
 	// Enable Logging : node name
 	telemetryNameCmd.Flags().StringVarP(&nodeName, "name", "n", "", "Friendly-name to use for node")
 	telemetryEndpointCmd.Flags().StringVarP(&uri, "endpoint", "e", "", "Endpoint's URI")
 }
 
+// If we didn't get a value from -d, try $ALGORAND_DATA
+func maybeUpdateDataDirFromEnv() {
```

Comment:
Nit: how about `resolveDataDir() string` which returns dataDir || os.Getenv()?  Updating the global var is ugly.  It looks like nil isn't really meaningful in calls to EnsureTelemetryConfig(), so you can just call EnsureTelemetryConfig(resolveDataDir(), ...).